Library/Homebrew/formula_installer.rb

```diff
@@ -625,9 +661,10 @@ def pour
       downloader.stage
     end
 
-    keg = Keg.new(formula.prefix)
-    keg.relocate_install_names Keg::PREFIX_PLACEHOLDER, HOMEBREW_PREFIX.to_s,
-      Keg::CELLAR_PLACEHOLDER, HOMEBREW_CELLAR.to_s, :keg_only => formula.keg_only?
+    if !pour_bottle? || formula.bottle.needs_relocation?
```

Comment:
Oh, you're absolutely right. That `pour_bottle?` is redundant. I'll remove it right now.

Looking at the code in `Keg#relocate_install_names`, it seems as if the only thing being done is changing the install names of binaries and no operations on any text files. Therefore, this new guard shouldn't affect the relocation of any non-binaries. Please correct me if I'm wrong :smile: